### PR TITLE
Update sorting out of starlark flags to also identify external repo starlark flags

### LINKED ISSUE #11506: clean command errors if Starlark-defined flags are provided
### Description of the problem / feature request:

I have set a Starlark-defined flag in my `.bazelrc` file:
```
build --@io_bazel_rules_go//go/config:incompatible_package_conflict_is_error=True
```

Since `bazel clean` inherits the command-line flags from `build`, running `bazel clean` fails with the following error message:
```
INFO: Options provided by the client:
  [...]
INFO: Reading rc options for 'clean' from <redacted>/.bazelrc:
  Inherited 'common' options: [...]
INFO: Reading rc options for 'clean' from <redacted>/.bazelrc:
  Inherited 'build' options: --@io_bazel_rules_go//go/config:incompatible_package_conflict_is_error=True  [...]
ERROR: Unrecognized arguments: --@io_bazel_rules_go//go/config:incompatible_package_conflict_is_error=True
```

It's unclear to me why `bazel clean` looks at any flags defined for `bazel build` (AFAICT, clean unconditionally deletes everything, so it probably shouldn't even look at `.bazelrc` `common` flags). At the very least, `bazel clean` should ignore Starlark-defined flags.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

clone `https://github.com/bazelbuild/rules_go`
run `bazel clean --@io_bazel_rules_go//go/config:incompatible_package_conflict_is_error=True`

### What operating system are you running Bazel on?

macOS

### What's the output of `bazel info release`?

3.1.0 / 3.2.0rc1

### PR DESCRIPTION
**Pull request by juliexxia**
Fixes #11506

### CODE REVIEW

File: src/test/java/com/google/devtools/build/lib/skylark/StarlarkOptionsParsingTest.java [hunk 1]
```diff
@@ -314,4 +319,16 @@ public void testOptionsAreParsedWithBuildTestsOnly() throws Exception {
 
     assertThat(result.getStarlarkOptions().get("//test:my_int_setting")).isEqualTo(15);
   }
+
+  @Test
+  public void testRemoveStarlarkOptionsWorks() throws Exception {
```
Comment (jin) [inline src/test/java/com/google/devtools/build/lib/skylark/StarlarkOptionsParsingTest.java:324]: Could you also add a test for the `--@//external/starlark/option` main repo case please?

File: src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java [hunk 2]
```diff
@@ -228,7 +228,7 @@ private Target loadBuildSetting(
     ImmutableList.Builder<String> keep = ImmutableList.builder();
     ImmutableList.Builder<String> remove = ImmutableList.builder();
     for (String name : list) {
-      ((name.startsWith("--//") || name.startsWith("--no//")) ? remove : keep).add(name);
+      (name.matches("--(no)?(@.*)?//.*") ? remove : keep).add(name);
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java:231]: Regex-based parsing is going to fall into difficulties. Can this just strip off the -- and the no (if present), and call [LabelValidator.validateAbsoluteLabel](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/cmdline/LabelValidator.java;drc=61d663a503710c8cb0d03d18f2eb776981b49e17;l=227), and check for exceptions being thrown?

Comment (juliexxia) [reply src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java]: So much better. Done.

File: src/test/java/com/google/devtools/build/lib/skylark/StarlarkOptionsParsingTest.java (hunk 1 above)
Comment (juliexxia) [reply src/test/java/com/google/devtools/build/lib/skylark/StarlarkOptionsParsingTest.java:324]: Yep! added

File: src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java [hunk 3]
```diff
@@ -228,7 +230,21 @@ private Target loadBuildSetting(
     ImmutableList.Builder<String> keep = ImmutableList.builder();
     ImmutableList.Builder<String> remove = ImmutableList.builder();
     for (String name : list) {
-      ((name.startsWith("--//") || name.startsWith("--no//")) ? remove : keep).add(name);
+      if (!name.startsWith("--")) {
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java:233]: This logic is great, but needs comments. More comments.

Comment (juliexxia) [reply src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java:239]: Done! added a description to the javadoc as well

**Pull request closed without merging**